api/src/scripts/translateResourceDescriptions.js

```diff
@@ -0,0 +1,126 @@
+ * you can mock out the resultant data to feed into Step 3
+ */
+const apiKey = '';
+// map ids of resources to translated descriptions
+var translatedResourceDescriptions = new Map();
+const languageType = process.argv.slice(2)[0]; // the language specified in the argument when running the script - the "target" argument in the POST request
+resourceDescriptions.forEach(function (resource) {
+  var id = resource._id;
+  var descriptionText = resource.description;
+  var source =
+    `https://www.googleapis.com/language/translate/v2?key=${apiKey}&source=en&target=${language}&callback=translateText&q=` +
+    descriptionText;
+  async (body) => {
+    const { resourceDescriptions } = body;
+    const res = await fetch(source, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        resourceDescriptions,
+      }),
+    });
+    return translatedResourceDescriptions.set(
+      id,
+      res.json().body.data.translations[0].translatedText, // store translated descriptions with id
+    );
+  };
+});
+
+/**
+ * Step 3: Store those translations back into our database
+ *
+ * You can look at our model for a translation - essentially, a "Translation" has a field "language" and then a field "messages" which is a Map of strings to other strings
+ * This is the "message ID" to its translation, so for example, if a resource had _id "1234", and translation "hola" for its description, after entering that into our db,
+ * the resulting db entry might look like:
+ * {
+ *  language: "es",
+ *  messages: {
+ *      "resource-description-6789": "gracias",
+ *      "resource-description-1234": "hola"
+ *  }
+ * }
+ *
+ * We want to loop over our results from Step 2, and enter all of those into our database in the format above ^
+ */
+
+var messagesMap = new Map();
+// store all descriptions with ids in messages map
+Object.keys(translatedResourceDescriptions).forEach(function (id) {
+  var messageKey = 'resource-description-' + id;
+  var messageValue = translatedResourceDescriptions.get(id);
+  messages.set(messageKey, messageValue);
+});
+
+var translation = new Translation({
+  language: languageType,
+  messages: messagesMap,
+});
+
+// Here we can use some update function on our model `Translation` (already imported) - you can look at the PUT request for /translation in api/admin.js for how to do this
+const AUTH_SERVER_URI = 'https://nawc-staging.vercel.app/auth';
+
+export const saveTranslation = () =>
+  fetch(`${AUTH_SERVER_URI}/api/admin/translation`, {
+    method: 'PUT',
+    headers: {
+      'Content-Type': 'application/json',
+      token: localStorage.getItem('token'),
+    },
+    body: JSON.stringify({
+      translationMessage: translation,
+    }),
+  })
+    .then((res) => res.json())
+    .catch((err) => {
+      console.err(err);
+      return null;
+    });
```

Comment:
Whoops, sorry I wasn't clear enough with my comment - I meant looking at [this](https://github.com/hack4impact-uiuc/ymca/blob/5d2ac23e613e412d52f1d6564ceca7a4cafa9c32/api/src/api/admin.js#L321) endpoint to see how we interact with the db.  We can basically copy lines 325-329, except rather than only `set`ting one message, we just loop through all of our messages and set them (and then save the model) 🥳 

What you did initially by trying to save the translation is a really good approach, but I'm trying to kind of account for if our translation model is already created for a specific language.

Reply:
Ohh that makes sense now, thank you!